gfx.cpp

```diff
@@ -9372,8 +9372,12 @@ class GfxInternal
         }
 
         uint64_t shader_key = 0;
-        std::string shader_key_bytecode;
-        std::string shader_key_reflection;
+        std::wstring shader_key_dir;
+        std::wstring shader_key_bytecode;
+        std::wstring shader_key_reflection;
+        std::wstring const shader_cache_dir = L"./shader_cache/";
+        std::wstring const shader_pdb_dir = L"./shader_pdb/";
+        constexpr uint32_t max_cached_files = 5;
```

Comment:
The `max_cached_files` variable is used to set the maximum number of cache/pdb files stored for a kernel before they are deleted. It is default set at 5 so that when changing options it doesnt always trigger a recompile but minimises the potential number of redundant cache/pdbs